components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/AbstractUserInfoResponseBuilder.java

```diff
@@ -79,14 +79,26 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
         filteredUserClaims.putAll(essentialClaims);
 
         //Handle essential claims of the request object
-        if (getRequestObject(tokenResponse) != null) {
-            filteredUserClaims.putAll(getUserClaimsFilteredByEssentialClaims(userClaims, USERINFO,
-                    getRequestObject(tokenResponse)));
-        }
+        filteredUserClaims.putAll(filterClaimsFromRequestObject(userClaims, getAccessToken(tokenResponse)));
 
         return buildResponse(tokenResponse, spTenantDomain, filteredUserClaims);
     }
 
+    private Map<String, Object> filterClaimsFromRequestObject(Map<String, Object> userAttributes,
+                                                              String token) throws OAuthSystemException {
+        try {
+            List<RequestedClaim> requestedClaims = OpenIDConnectServiceComponentHolder.getRequestObjectService().
+                    getRequestedClaimsForUserInfo(token, null);
+            return OpenIDConnectServiceComponentHolder.getInstance()
+                    .getHighestPriorityOpenIDConnectClaimFilter()
+                    .getClaimsFilteredByEssentialClaims(userAttributes, requestedClaims);
```

Comment:
@madurangasiriwardena Do you know the usage of getClaimsFilteredByEssentialClaims()